16/umbraco-cms/customizing/extending-overview/extension-types/kind.md

```diff
@@ -80,32 +78,41 @@ In this example, the Header App extension uses the `kind: 'button'`, meaning it
 
 Here’s an example of how to register and use the Button Kind in a Header App extension:
 
+{% code title="kinds/manifests.ts" %}
+
+{% hint style="info" %}
+This example uses the dynamic extension registration approach. `umbExtensionsRegistry.register()` might be called from within an entrypoint lifecycle method like `onInit()`. For more information, see the [Backoffice Entry Point](./backoffice-entry-point.md) article.
+{% endhint %}
+
 ```typescript
 import { umbExtensionsRegistry } from '@umbraco-cms/backoffice/extension-registry';
+import type { UmbExtensionManifestKind } from "@umbraco-cms/backoffice/extension-registry";
 
-const manifest: UmbExtensionManifest = {
+export const customHeaderAppButton: UmbExtensionManifestKind = {
   type: 'kind',
   alias: 'Umb.Kind.MyButtonKind',  // Alias for the Kind
   matchType: 'headerApp', // Extension type the Kind applies to
-  matchKind: 'button',  // Defines the Kind alias
+  matchKind: 'customHeaderAppButton',  // Defines the Kind alias
   manifest: {
     elementName: 'umb-header-app-button',
   },
 };
 
 umbExtensionsRegistry.register(manifest);
 ```
+{% endcode %}
 
-This code registers the Button Kind, so other Header App extensions using `type: 'headerApp'` and `kind: 'button'` will inherit the preset `elementName: 'umb-header-app-button'`.
+This code registers the Button Kind, so other Header App extensions using `type: 'headerApp'` and `kind: 'customHeaderAppButton'` will inherit the preset `elementName: 'umb-header-app-button'`.
```

Comment:
Again here, the `umb-header-app-button` element is the Core element for the Core Kind called 'button' for Ext Type of 'HeaderApp'. so what it said before is correct. But there maybe something unclear in the story.Maybe it has to be spelled out that this is using a kind from the Core/System?